library/Icingadb/Model/CustomvarFlat.php

```diff
@@ -97,7 +97,8 @@ public function unFlattenVars(Traversable $flattenedVars): array
             $step = array_shift($path);
 
             $isIndex = (bool) preg_match('/^\[(\d+)]$/', $step, $m);
-            if ($isIndex) {
+            // Don't compute custom var names like `vars."[4]"` as array index access.
+            if (! isset($source[$step]) && $isIndex) {
```

Comment:
Then you don't have to perform the preg_match, do you?

```suggestion
            $isIndex = ! isset($source[$step]) && (bool) preg_match('/^\[(\d+)]$/', $step, $m);
            if ($isIndex) {
```